src/models/graph.ts

```diff
@@ -378,6 +378,27 @@ export class Graph<N extends INodeBase, E extends IEdgeBase> extends Subject imp
   }
 
   update(data?: IObserverDataPayload): void {
+    if (data && 'type' in data && 'options' in data && 'isSingle' in data.options) {
+      if (data.type === 'node' && data.options.isSingle) {
+        const nodes = this._nodes.getAll();
+
+        for (let i = 0; i < nodes.length; i++) {
```

Comment:
I feel like these iterations through all nodes/edges can be slow. Any ideas? And do we want to clear the state of all nodes/edges or only the adjusting ones?

Reply:
Or maybe to supplement this nodes array we can keep an index where `nodeIndex[node.id] = nodes.findIndex((el) => el.id === node.id)`

Reply:
Are you suggesting storing nodes in a hashmap? As far as I know, retrieving elements from an array by a known index is `O(1)` complexity, so iterating through an array vs a hashmap (which also has `O(1)` complexity for retrieving element by key) may not make a big difference here. I might have misunderstood your point, or perhaps I missed something, so feel free to clarify it further :)

Reply:
If all nodes are in state X, and you want to clear them up if they are not node with ID 1, you still need to to `N - 1` updates. So hash maps or any other structure won't help there, you already have the fastest structure, an array.

But, you can do this and increase the memory footprint: I am not sure if it makes sense to go into this right now, but this would be the plan:
- Graph keeps a structure of nodes/edges by state, so each state has a structure of nodes and edges (probably just id) - a set of a map.
- States in nodes and edges are just views, readonly
- When state is updated in node, edge it is just updated in the graph

This way, if you have `isSingle: true`, graph, as an owner of the data can then literally take all selected nodes from a set or a map, clear those states and set the new node as single selected. This would improve the speed, but it would worse the memory footprint and code simplitiy because owner of the state becomes the graph, not graph objects (node/edge).

I would keep this as is and fix only if we see that it hurts performance.

Reply:
Sorry, I misunderstood the functionality. I saw `===` instead of `!==`. I thought you were trying to clear the state of anode with a specific `id`.